### ISSUE TITLE
Ruff flags (and tries to fix) UP007 for python<3.10 in some cases

### ISSUE THREAD

**Issue by FranzForstmayr** [astral-sh/ruff#6313]
Following code works without error when checked with
`ruff .\test.py --isolated --select UP --target-version py39`
```python
# from __future__ import annotations

from typing import Union

class MyModel():
    value: Union[bool, str]
```

When I uncomment the first line, `ruff` reports (and wants to fix) the `Union[bool, str]` to `bool | str` which is not compatible with python 3.9.

```python
from __future__ import annotations

from typing import Union

class MyModel():
    value: Union[bool, str]
```
Same command
`ruff .\test.py --isolated --select UP --target-version py39`

```
test.py:6:12: UP007 [*] Use `X | Y` for type annotations
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

**Comment by charliermarsh** [comment]
I believe that code _is_ Python 3.9-compatible -- it will run without error (I just tried on 3.8), and all type checkers will support and understand it. `__future__` annotations effectively treats all annotations as strings, so analysis tools can support language features that don't exist at runtime.

If you're relying on the annotations being _evaluated_ at runtime, for use with Pydantic or some other library, you may be looking for the [`keep-runtime-typing`](https://beta.ruff.rs/docs/settings/#pyupgrade-keep-runtime-typing) setting.

**Comment by FranzForstmayr** [comment]
Yes, this is a snippet of a bigger `pydantic` based model where I got the error. 
You are right, python3.9 can run this snippet without error, but based on the error message I thought this is not supported yet.

[PEP604](https://peps.python.org/pep-0604/) mentions also only 3.10+.

Thanks anyway, `keep-runtime-typing` will solve my issue :)
